hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -1284,6 +1288,7 @@ void startCommonServices(Configuration conf, HAContext haContext) throws IOExcep
       dir.setINodeAttributeProvider(inodeAttributeProvider);
     }
     snapshotManager.registerMXBean();
+    snapshotDeletionGc.schedule();
```

Comment:
I think its better to start the gc work in FSNameSystem#startActiveServices() after quota setup and initialization is done

startActiveServices()
``
 // Initialize the quota.
      dir.updateCountForQuota();
      // Enable quota checks.
      dir.enableQuotaChecks();

Reply:
Sure.  Thanks for the suggestion.